api/pkg/apis/projectcalico/v3/bgpconfig.go

```diff
@@ -91,6 +91,10 @@ type BGPConfigurationSpec struct {
 	// This field can only be set on the default BGPConfiguration instance and requires that NodeMesh is enabled
 	// +optional
 	NodeMeshMaxRestartTime *metav1.Duration `json:"nodeMeshMaxRestartTime,omitempty" confignamev1:"node_mesh_restart_time"`
+
+	// BindMode indicates whether to listen for BGP connections on all addresses (None)
+	// or only on the node's canonical IP address Node.Spec.BGP.IPvXAddress (NodeIP).
+	BindMode string `json:"bindMode,omitempty" validate:"omitempty,oneof=None NodeIP"`
```

Comment:
For API fields like this, we should be defining a new type. e.g.,

```go
type BindMode string

const (
    BindModeNone BindMode = "None"
    BindModeNodeIP BindMOde = "NodeIP"
)
```

Since this is an optional field, per Kubernetes API conventions, this should also be a pointer and be marked as optional.

e.g.,

```go
// +optional
BindMode *BindMode `json:"bindMode,omitempty"`
```

Reply:
The comment should also include the default behavior for the reader's benefit.

Reply:
I'm afraid I gave @hanamantagoudvk the opposite feedback, earlier on this PR.  @caseydavenport Are we sure that the additional boilerplate for **not** using `oneof=` is really useful?

Reply:
I think stylistically it can come down to preference, but this comes straight from the Kubernetes API conventions doc which we try to follow best we can: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types